Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/VaccinatedPersonTest.kt

```diff
@@ -112,20 +125,119 @@ class VaccinatedPersonTest : BaseTest() {
+            // Last day before immunity, UI shows 0 days until immunity
+            Instant.parse("2021-06-27T12:00:00.000Z").let { now ->
+                getTimeUntilImmunity(now)!!.apply {
+                    standardHours shouldBe 10 // In 10 hours it is 22 UTC, which is midnight in the users TZ.
+                    standardDays shouldBe 0
+                    this shouldBe Duration.standardHours(10)
+                }
+                getVaccinationStatus(now) shouldBe VaccinatedPerson.Status.COMPLETE
+            }
+            // Immunity should be reached at midnight in the users timezone
+            Instant.parse("2021-06-27T22:00:00.000Z").let { now ->
+                getTimeUntilImmunity(now)!!.apply {
+                    standardHours shouldBe 0
+                    standardDays shouldBe 0
+                    this shouldBe Duration.ZERO
+                }
+                getVaccinationStatus(now) shouldBe VaccinatedPerson.Status.IMMUNITY
+            }
+        }
+    }
+
+    @Test
+    fun `time until immunity - case Luka#1`() {
+        DateTimeZone.setDefault(DateTimeZone.forTimeZone(TimeZone.getTimeZone("Europe/Berlin")))
+
+        val personData = mockk<VaccinatedPersonData>().apply {
+            every { vaccinations } returns setOf(
+                mockk<VaccinationContainer>().apply {
+                    every { toVaccinationCertificate(any()) } returns mockk<VaccinationCertificate>().apply {
+                        every { vaccinatedAt } returns LocalDate.parse("2021-01-01")
+                        every { doseNumber } returns 2
+                        every { totalSeriesOfDoses } returns 2
+                    }
+                }
+            )
+        }
+
+        VaccinatedPerson(data = personData, valueSet = null).apply {
+            Instant.parse("2021-01-14T0:00:00.000Z").let { now ->
+                getTimeUntilImmunity(now)!!.standardHours shouldBe 47
+                getTimeUntilImmunity(now)!!.standardDays shouldBe 1
+                getVaccinationStatus(now) shouldBe VaccinatedPerson.Status.COMPLETE
+            }
+            Instant.parse("2021-01-15T0:00:00.000Z").let { now ->
+                getTimeUntilImmunity(now)!!.standardHours shouldBe 23
+                getTimeUntilImmunity(now)!!.standardDays shouldBe 0
+                getVaccinationStatus(now) shouldBe VaccinatedPerson.Status.COMPLETE
+            }
+            // Case Luka#1 happens on 01.15.21, this mean it's winter time!
+            // The users timezone is GMT+1 (winter-time) (UTC+1), not GMT+2 (summer-time) (UTC+2)
+            Instant.parse("2021-01-15T22:00:00.000Z").let { now ->
+                getTimeUntilImmunity(now)!!.standardMinutes shouldBe 60
+                getTimeUntilImmunity(now)!!.standardHours shouldBe 1
+                getTimeUntilImmunity(now) shouldBe Duration.standardHours(1)
+                getVaccinationStatus(now) shouldBe VaccinatedPerson.Status.COMPLETE
+            }
```

Comment:
@d4rken `"2021-01-15T23:00:00.000Z"` should then yield `IMMUNITY`, right? Can you add that case?

Reply:
added